### PR TITLE
[OLD] Add SonarCloud with gradle.properties

### PR DESCRIPTION
**Pull request by marko-radosavljevic**
This PR will configure SonarCloud. :tada: 

We integrate Sonar scanner into our gradle build. That is the recommended way of configuring SonarCloud for our codebase.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import com.github.vlsi.gradle.properties.dsl.props
 plugins {
     `java-library`
     application
+    id("org.sonarqube") version "3.0"
```
Comment (weisJ) [inline build.gradle.kts]: The version should go into `gradle.properties` together with the id declaration in `settings.gradle.kts`

**Comment by weisJ** [comment]
Does sonarcloud get run by `build` etc by default? If yes then the task needs to be disabled by default and only be enabled on the github workflow. Not everyone has access to the login token.

**Comment by alehel** [comment]
@marko-radosavljevic I don't see any config for sonarcloud in the github workflows. Has this been configured?

**Comment by alehel** [comment]
@weisJ not by default no. Only if you specify that it should run, so that shouldn't be a problem.

**Comment by marko-radosavljevic** [comment]
Yup, sorry, busy day.

Thanks for the input, i will implement this fully later today, its a WIP ^^

Will also edit my PR to add more details, for documentation purposes, and of course, I will ask for a review when I'm done.

**Comment by marko-radosavljevic** [comment]
Closing this one because it's outdated. New PR was originally created to test if creating PR from upstream branch resolved our issues with SonarCloud auth (it did).  I will merge form there because it's much cleaner and more accurate PR compared to this one.

**Pull request closed without merging**